kafka_consumer_freshness_tracker/src/test/java/com/tesla/data/consumer/freshness/ConsumerFreshnessTest.java

```diff
@@ -272,6 +277,124 @@ public void testBurrowMissingConsumerGroupPartitionEndOffsetMarksMissing() throw
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "bad_cluster";
+    when(burrow.getClusterBootstrapServers(clusterName)).thenThrow(new IOException());
+    Map<String, Object> conf = mockConfForCluster(
+            clusterName, "l1.example.com:9092", "l2.example.com:9092"
+    );
+
+    ConsumerFreshness freshness = new ConsumerFreshness();
+    // A runtime exception should occur when the burrow call returns 404
+    thrown.expect(RuntimeException.class);
+    freshness.setupWithBurrow(conf, burrow);
+  }
+
+  @Test
+  public void testNormalModeConfigurationIsValidServersMatchBurrow() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterBootstrapServers(clusterName))
+            .thenReturn(Arrays.asList("kafka01.example.com:10251", "kafka02.example.com:10251")
+      );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+        "kafka01.example.com:10251", "kafka02.example.com:10251", "kafka03.example.com:10251"
+        );
+
+    ConsumerFreshness freshness = new ConsumerFreshness();
+
+    // This exception should be thrown when the consumers are attempted to be
+    // connected after validation succeeds
+    thrown.expect(KafkaException.class);
+    thrown.expectMessage("Failed to construct kafka consumer");
+
+    freshness.setupWithBurrow(conf, burrow);
+  }
+
+  @Test
+  public void testNormalModeConfigurationIsInvalidContainsUnknownServer() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterBootstrapServers(clusterName))
+      .thenReturn(Arrays.asList(
+            "kafka01.example.com:10251", "kafka02.example.com:10251")
+          );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+        "kafka01.example.com:10251", "kafka02.example.com:10251", "kafka03.example.com:10251"
+        );
+
+    ConsumerFreshness freshness = new ConsumerFreshness();
+
+    // A runtime exception should occur when the burrow call returns 404
+    thrown.expect(RuntimeException.class);
+    freshness.setupWithBurrow(conf, burrow);
+  }
+
+  @Test
+  public void testStrictModeConfigurationIsValidServersMatchBurrow() throws Exception {
+    Burrow burrow = mock(Burrow.class);
+    String clusterName = "cluster1";
+    when(burrow.getClusterBootstrapServers(clusterName))
+      .thenReturn(Arrays.asList("kafka01.example.com:10251", "kafka02.example.com:10251")
+          );
+
+    Map<String, Object> conf = mockConfForCluster(
+        clusterName,
+            "kafka01.example.com:10251", "kafka02.example.com:10251"
+        );
+
+    ConsumerFreshness freshness = new ConsumerFreshness();
+    freshness.strict = true;
+
+    // This exception should be thrown when the consumers are attempted to be
+    // connected after validation succeeds
+    thrown.expect(KafkaException.class);
```

Comment:
This is a very strange way to test a "successful" validation. Can we restructure the validation (maybe just expose that bit) to make this feel more natural?

Reply:
fixed